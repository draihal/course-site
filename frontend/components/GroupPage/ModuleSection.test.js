import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestModuleSection from './ModuleSection'

const modules = [
  {
    "id": 1,
    "lessons": [],
    "name": "1 месяц",
  },
  {
    "id": 2,
    "lessons": [],
    "name": "2 месяц",
  },
  {
    "id": 3,
    "lessons": [],
    "name": "3 месяц",
  }
];



describe('ModuleSection', () => {
  it('render correctly ModuleSection component', () => {
    const tree = renderer.create(<TestModuleSection modules={modules} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render ModuleSection correctly with given props', () => {
    const { queryByText } = render(<TestModuleSection modules={modules} />);
    expect(queryByText(/1 месяц/i)).toBeTruthy();
    expect(queryByText(/2 месяц/i)).toBeTruthy();
    expect(queryByText(/3 месяц/i)).toBeTruthy();
  });

});