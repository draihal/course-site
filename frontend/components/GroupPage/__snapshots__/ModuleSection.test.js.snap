// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleSection render correctly ModuleSection component 1`] = `
<div
  className="col-md-12"
>
  <div>
    <h3>
      Модуль: 
      1 месяц
    </h3>
    <hr />
    <div
      className="col-md-12"
    />
  </div>
  <div>
    <h3>
      Модуль: 
      2 месяц
    </h3>
    <hr />
    <div
      className="col-md-12"
    />
  </div>
  <div>
    <h3>
      Модуль: 
      3 месяц
    </h3>
    <hr />
    <div
      className="col-md-12"
    />
  </div>
</div>
`;
